File: 09_DataStructure_Algorithm/0513_day48/02_네트워크/네트워크_dfs.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] visited = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                answer++;
                dfs(i, computers, visited);
            }
        }
        
        return answer;
    }
    
    void dfs(int cur, int[][] computers, boolean[] visited) {
        visited[cur] = true;
        
        for (int nxt = 0; nxt < computers[cur].length; nxt++) {
            if (!visited[nxt] && computers[cur][nxt] == 1) {
                dfs(nxt, computers, visited);
            }
        }
    }
}

File: 09_DataStructure_Algorithm/0513_day48/02_네트워크/네트워크_dfs.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    def dfs(cur):
        visited[cur] = True
        
        for nxt in range(n):
            if not visited[nxt] and computers[cur][nxt] == 1:
                dfs(nxt)
    
    for i in range(n):
        if not visited[i]:
            answer += 1
            dfs(i)
    
    return answer